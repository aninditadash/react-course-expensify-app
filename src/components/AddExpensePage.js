import React from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";

import ExpenseForm from "./ExpenseForm";
import { startAppExpense } from "../actions/expenses";

export const AddExpensePage = (props) => {
  const navigate = useNavigate();

  const onSubmit = (expense) => {
    props.startAppExpense(expense);
    navigate("/");
  };

  return (
    <>
      <div className="page-header">
        <div className="content-container">
          <h1 className="page-header__title">Add Expense</h1>
        </div>
      </div>
      <div className="content-container">
        <ExpenseForm onSubmit={onSubmit} />
      </div>
    </>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startAppExpense: (expense) => dispatch(startAppExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
