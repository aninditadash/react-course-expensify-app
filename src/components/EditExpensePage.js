import React from "react";
import { connect } from "react-redux";
import { useParams, useNavigate } from "react-router-dom";

import ExpenseForm from "./ExpenseForm";

import {
  editExpense,
  startRemoveExpense,
  startEditExpense
} from "../actions/expenses";

export const EditExpensePage = (props) => {
  let { id } = useParams();
  const navigate = useNavigate();
  const selectedExpense = props.expenses.find((expense) => expense.id === id);

  const onSubmit = (expense) => {
    props.startEditExpense(id, expense);
    navigate("/");
  };

  const onRemove = () => {
    props.startRemoveExpense({ id });
    navigate("/");
  };

  return (
    <div>
      <ExpenseForm expense={selectedExpense} onSubmit={onSubmit} />
      <button onClick={onRemove}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses
  };
};

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
