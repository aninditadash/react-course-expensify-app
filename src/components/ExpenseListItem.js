import React from "react";
import moment from "moment";
import numeral from "numeral";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="expense-list-item" to={`/edit/${id}`}>
    <div>
      <p className="expense-list-item__title expense-list-item--bold">
        {description}
      </p>
      <p className="expense-list-item__subtitle">
        {moment(createdAt).format("MMMM Do, YYYY")}
      </p>
    </div>
    <p className="expense-list-item__data expense-list-item--bold">
      {numeral(amount).format("$0,0.00")}
    </p>
  </Link>
);

export default ExpenseListItem;
