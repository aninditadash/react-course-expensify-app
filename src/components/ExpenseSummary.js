import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          {expenseCount > 1 ? (
            <p className="page-header__title-bold">
              Viewing {expenseCount} expenses totalling{" "}
              <span>{numeral(expensesTotal).format("$0,0.00")}</span>
            </p>
          ) : (
            <p>
              Viewing {expenseCount} expense totalling{" "}
              <span>{numeral(expensesTotal).format("$0,0.00")}</span>
            </p>
          )}
        </h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">
            Add Expense
          </Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: getExpensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpenseSummary);
