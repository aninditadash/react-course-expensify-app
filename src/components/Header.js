import React from "react";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import { startLogout } from "../actions/auth";

// import { history } from "../routers/AppRouter";

// import history from "../history";

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>
    <NavLink
      to="/dashboard"
      className={({ isActive }) => (isActive ? "is-active" : undefined)}
    >
      Dashboard
    </NavLink>
    <NavLink
      to="/create"
      className={({ isActive }) => (isActive ? "is-active" : undefined)}
    >
      Create Expense
    </NavLink>
    <NavLink
      to="/edit"
      className={({ isActive }) => (isActive ? "is-active" : undefined)}
    >
      Edit Expense
    </NavLink>
    <NavLink
      to="/help"
      className={({ isActive }) => (isActive ? "is-active" : undefined)}
    >
      Help
    </NavLink>
    <button onClick={startLogout}>Logout</button>
  </header>
);

const mapStateToProps = (dispatch) => ({
  startLogout: () => {
    // history.push("/");
    return dispatch(startLogout());
  }
});

export default connect(undefined, mapStateToProps)(Header);
