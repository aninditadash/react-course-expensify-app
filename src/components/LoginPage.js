import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
// import { useNavigate } from "react-router-dom";

// import history from "../history";
// import { store } from "../app";

// import { history } from "../routers/AppRouter";

export const LoginPage = ({ startLogin }) => {
  // const navigate = useNavigate();

  //   const onSubmit = (expense) => {
  //     props.startAppExpense(expense);
  //     navigate("/");
  //   };
  // debugger;

  // const login = () => {
  //   startLogin();
  //   // navigate("/dashboard");
  // };

  return (
    <div>
      Login Page
      <button onClick={startLogin}>Login</button>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
  // {
  //   // store.dispatch(startLogin()).then(() => {
  //   //   render();
  //   //   if (history.location.pathname === "/") {
  //   //     history.push("/dashboard");
  //   //   }
  //   // });
  //   // navigate("/dashboard");
  //   return .then(() => {
  //     // debugger;
  //     // navigate("/dashboard");
  //     // history.push("/dashboard");
  //   });
  // }
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
