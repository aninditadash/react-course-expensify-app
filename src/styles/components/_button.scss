.button {
  background: $button-blue;
  color: white;
  outline: none;
  border: none;
  font-weight: 300;
  padding: $s-size;
  font-size: 1.8rem;
  text-decoration: none;
  display: inline-block;
  line-height: 1;
}

.button__link {
  background: none;
}

.button--secondary {
  background: #888888;
}
