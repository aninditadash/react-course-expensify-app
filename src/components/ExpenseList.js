import React from "react";
import { connect } from "react-redux";

import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";

export const ExpenseList = (props) => (
  <div className="content-container">
    <div className="expense-list__header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
    </div>
    <div className="expense-list__body">
      {props.expenses.length === 0 ? (
        <div className="expense-list-item  expense-list__message">
          No expenses
        </div>
      ) : (
        props.expenses.map((expense) => (
          <ExpenseListItem key={expense.id} {...expense} />
        ))
      )}
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
