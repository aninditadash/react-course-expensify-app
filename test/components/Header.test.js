import React from "react";
import { shallow } from "enzyme";
import { Header } from "../../src/components/Header";

// const { createBrowserHistory } = require("history");
import { createBrowserHistory } from "history";
import { Router } from "react-router-dom";

// const history = createBrowserHistory({ forceRefresh: true });

// sinon.spy(history, "push");

let startLogout, wrapper;

// const id = "2";
// const createBrowserHistoryFn = jest.fn();
// const mockedUsedParams = { id };

// jest.mock("history", () => ({
//   ...jest.requireActual("history"),
//   createBrowserHistory: () => createBrowserHistoryFn
//   // useParams: () => mockedUsedParams
// }));

beforeEach(() => {
  startLogout = jest.fn();
  wrapper = shallow(<Header startLogout={startLogout} />);
});

test("Header Component: Should render Header correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("Header Component: Should call startLogout on button click", () => {
  wrapper.find("button").simulate("click");
  expect(startLogout).toHaveBeenCalled();
});
