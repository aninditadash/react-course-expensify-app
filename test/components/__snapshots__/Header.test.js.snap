// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component: Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    className={[Function]}
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    className={[Function]}
    to="/create"
  >
    Create Expense
  </NavLink>
  <NavLink
    className={[Function]}
    to="/edit"
  >
    Edit Expense
  </NavLink>
  <NavLink
    className={[Function]}
    to="/help"
  >
    Help
  </NavLink>
  <button
    onClick={[MockFunction]}
  >
    Logout
  </button>
</header>
`;
